src/librustc/mir/interpret/error.rs

```diff
@@ -137,17 +137,18 @@ impl<'tcx> ConstEvalErr<'tcx> {
         message: &str,
         lint_root: Option<hir::HirId>,
     ) -> Result<DiagnosticBuilder<'tcx>, ErrorHandled> {
+        let mut must_error = false;
         match self.error {
             err_inval!(Layout(LayoutError::Unknown(_))) |
             err_inval!(TooGeneric) =>
                 return Err(ErrorHandled::TooGeneric),
-            err_inval!(Layout(LayoutError::SizeOverflow(_))) |
             err_inval!(TypeckError) =>
                 return Err(ErrorHandled::Reported),
+            err_inval!(Layout(LayoutError::SizeOverflow(_))) => must_error = true,
```

Comment:
The main error message will be bogus. Please overwrite `message`, too.

Reply:
done